karrot/conversations/api.py

```diff
@@ -300,14 +300,28 @@ def get_queryset(self):
             return qs.only_threads_with_user(self.request.user) \
                 .prefetch_related('participants', 'latest_message')
 
-        if self.action == 'list':
-            qs = qs.prefetch_related('reactions', 'participants', 'images')
-
         if self.request.query_params.get('thread', None):
             return qs.only_threads_and_replies()
 
         return qs.exclude_replies()
 
+    def list(self, request, *args, **kwargs):
+        # Workaround to avoid extremely slow cases: split up query
+        # 1. get message ids, including costly access control
+        # 2. get data, including costly annotations
```

Comment:
Maybe include a link to the issue too, in case in the future it becomes untrue (query planner can improve), and then the trail will be possible to follow.